test/test_csvs.py

```diff
@@ -0,0 +1,289 @@
+# OOIPLACEHOLDER
+#
+# Copyright 2018 Raytheon Co.
+
+import os
+import unittest
+from nose.plugins.attrib import attr
+
+@attr('UNIT')
+class GeneralUnitTest(unittest.TestCase):
+    
+    def _getPath(self, filename):
+        return os.path.join(os.path.dirname(__file__), filename)
+
+    def _getParameters(self):
+        parameters = {} # dictionary: key=name; value=data type number
+        with open(self._getPath('parameter.txt'), mode='rt') as f:
+            for line in f:
+                parts = line.strip().split(':')
+                parameters[parts[0]] = parts[1]
+        return parameters
+
+    def _getRefDesignators(self):
+        refdes = set()
+        with open(self._getPath('refdes.txt'), mode='rt') as f:
+            for line in f:
+                refdes.add(line.strip())
+        return refdes
+
+    def _getTypes(self):
+        types = {} # dictionary: key=number; value=data type
+        with open(self._getPath('type.txt'), mode='rt') as f:
+            for line in f:
+                parts = line.strip().split(':')
+                types[parts[0]] = parts[1]
+        return types
+
+    def _getUnits(self):
+        units = set()
+        units.add('') # empty value is valid
+        with open(self._getPath('unit.txt'), mode='rt') as f:
+            for line in f:
+                units.add(line.strip())
+        return units
+
+    def _isEmptyOrNumeric(self, value):
+        # if it's empty it's ok
+        if len(value) == 0: return True
+
+        # remove leading minus sign if it exists
+        if value[:1] == '-': value = value[1:]
+
+        # if there's only one decimal point remove it
+        if value.count('.') == 1:
+            value = value[:value.index('.')] + value[value.index('.')+1:]
+
+        return value.isdigit()
+
+    def _getValidationCollection(self):
+        units = self._getUnits()
```

Comment:
Again - we see here why this is not a good idea - you are loading the file with each and every call to _getUnits(). This should be done once in the unitTest setup (using more reliable python csv reading instead of home-grown bash parsing).

Reply:
Agreed as stated earlier.